src/python/pants/backend/go/util_rules/pkg_analysis.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
```

Comment:
Rather than `pkg_analysis`, can this be incorporated into `FirstPartyPkgInfo`? Iiuc, we only ever need to analyze first-party packages. We will continue to use `go list` for third-party, at least for now. (If that ever changes, we can of course rename to something more generic like `pkg_analysis.py`)

I believe your `AnalyzedGoPackage` is almost the same as `FirstPartyPkgInfo`.

Reply:
It is a superset. Do you still want to keep the minimal `FirstPartyPkgInfo`?

Frankly, I don't want to have to revisit this analysis code later to add back things (like `cgo_fles`) that I could have included now. I want it to match what `go list` would have done. I could also have the Go code export values that Python doesn't decode. Thoughts?

Reply:
> I could also have the Go code export values that Python doesn't decode.

I recommend having those unused fields be private with `_cgo_files`. That way you can test that your analysis works with Python tests, but production knows to not use them.

Reply:
> I recommend having those unused fields be private with `_cgo_files`. That way you can test that your analysis works with Python tests, but production knows to not use them.

Makes sense. Will do that.